### PR TITLE
HIVE-27409: Iceberg: table with EXTERNAL type can not use statistics to optimize the query - ADDENDUM

### PR DESCRIPTION
**Pull request by zhangbutao**
### Why are the changes needed?

I missed a fix when refining HIVE-19335 to enable EXTERNAL iceberg table use stats.  This was found when i rerun dynamic_semijoin_reduction.q in which i transform iceberg type to EXTERNAL, but actually the execution plan should not be changed. Please see https://github.com/apache/hive/pull/4341/files#diff-3abb3774d1aa95235ca6c87cae91452ab59ddce60b6270eb46c038a19d12d88a

### Does this PR introduce _any_ user-facing change?

NO

### CODE REVIEW

**Comment by zhangbutao** [comment]
@ayushtkn @akshat0395 @simhadri-g Sorry for previous missing fix. Please take a look again if having times.

**Review by ayushtkn — APPROVED** [review]
LGTM pending tests

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/DynamicPartitionPruningOptimization.java [hunk 1]
```diff
@@ -319,8 +319,9 @@ private boolean disableSemiJoinOptDueToExternalTable(HiveConf conf, TableScanOpe
           if (columnOrigin != null && columnOrigin.op instanceof TableScanOperator) {
             // Join key origin has been traced to a table column. Check if the table is external.
             TableScanOperator joinKeyTs = (TableScanOperator) columnOrigin.op;
-            if (MetaStoreUtils.isExternalTable(joinKeyTs.getConf().getTableMetadata().getTTable())) {
-              LOG.debug("Join key {} is from {} which is an external table. Disabling semijoin optimization.",
+            if (!StatsUtils.checkCanProvideStats(new Table(joinKeyTs.getConf().getTableMetadata().getTTable()))) {
+              LOG.debug("Join key {} is from {} which is an external table and also could not provide statistics. " +
```
Comment (aturoczy) [inline ql/src/java/org/apache/hadoop/hive/ql/optimizer/DynamicPartitionPruningOptimization.java:323]: I know It was in the previous version as well, but inside the { } should not be a string for the table name?

Comment (zhangbutao) [reply ql/src/java/org/apache/hadoop/hive/ql/optimizer/DynamicPartitionPruningOptimization.java:323]: It is a string placeholder for the table name indeed.  Table name value is here line  `L325 `https://github.com/apache/hive/pull/4395/files#diff-1ade8830fed6939060483b84496a26eecab0e7ef64e19265a8e1102e1178fea2L325

**Review by akshat0395 — APPROVED** [review]

**Comment by zhangbutao** [comment]
Added commit https://github.com/apache/hive/pull/4395/commits/24dc46f7738472b094191fbc94387840d511ae9f to fix failed qtests http://ci.hive.apache.org/blue/organizations/jenkins/hive-precommit/detail/PR-4395/1/tests/ which have EXTERNAL iceberg table. 

`mvn test -Dtest.output.overwrite -Dtest=TestIcebergCliDriver -Dqfile="delete_iceberg_partitioned_avro.q,delete_iceberg_partitioned_orc.q,delete_iceberg_partitioned_parquet.q,delete_iceberg_unpartitioned_parquet.q,update_iceberg_partitioned_avro.q,update_iceberg_partitioned_orc.q,update_iceberg_partitioned_parquet.q,update_iceberg_unpartitioned_parquet.q" -pl itests/qtest-iceberg -Piceberg -Pitests -Drat.skip`

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/DynamicPartitionPruningOptimization.java (hunk 1 above)
Comment (aturoczy) [reply ql/src/java/org/apache/hadoop/hive/ql/optimizer/DynamicPartitionPruningOptimization.java:322]: ACK In C# we have different style and it confused me :)

**Review by aturoczy — APPROVED** [review]
+1

**Pull request merged**